src/parser.c

```diff
@@ -1260,6 +1260,20 @@ void read_lvalue(lvalue_t *lvalue,
     while (lex_peek(T_open_square, NULL) || lex_peek(T_arrow, NULL) ||
            lex_peek(T_dot, NULL)) {
         if (lex_accept(T_open_square)) {
+            /* if previous operator requires deference, deference here, then
+             * offset later */
```

Comment:
Improve the descriptions.

Reply:
I just tested additional expressions e.g. "data.raw[0]" to better conclude this patch's effect, apperently this does also fix deference behavior on mixed subscript and dot operator (which also produced incorrect code before).